core/chain_makers.go

```diff
@@ -166,16 +166,21 @@ func GenerateChain(parent *types.Block, db common.Database, n int, gen func(int,
 }
 
 func makeHeader(parent *types.Block, state *state.StateDB) *types.Header {
-	time := parent.Time() + 10 // block time is fixed at 10 seconds
+	time := new(big.Int)
+	if parent.Time() == nil {
+		time = big.NewInt(10)
+	} else {
+		time = time.Add(parent.Time(), big.NewInt(10)) // block time is fixed at 10 seconds
```

Comment:
I think this above block could be a little cleaner as:

```
var time *big.Int
if parent.Time() == nil {
	time = big.NewInt(10)
} else {
	time = new(big.Int).Add(parent.Time(), big.NewInt(10)) // block time is fixed at 10 seconds
}
```

I.e.
 * don't create a `big.Int` first which will be destroyed anyway (i.e. one less memory alloc and free)
 * `time := time.Add` is redundant, as `Add` already updates `time`. However, the cleaner use is to work with a clean new int.